src/mlpack/methods/ann/activation_functions/elish_function.hpp

```diff
@@ -30,10 +30,90 @@
 namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
+/**
+ * The ELiSH function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     \frac{x}{1+e^{-x}} & : x \ge 0 \\
+ *     \frac{e^x-1}{1+e^{-x}} & : x < 0
+ *   \end{array}
+ * \right \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     \frac{1}{1+e^{-x}} + 
+ *              \frac{xe^{-x}}{ \left(1+e^{-x}}\right)^2} & : x \ge 0 \\
+ *     e^x-\frac{2}{1+e^x}+\frac{2}{\left(1+e^x}\right)^2} & : x < 0
+ *   \end{array}
+ * \right.
+ * @f}
+ */
 class ElishFunction
 {
  public:
+ /**
+   * Computes the ELiSH function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+    if (x < 0.0) {
+        return (std::exp(x) - 1) / (1 + std::exp(-x));
+    }
+    return x / (1 + std::exp(x));
+  }
+
+  /**
+   * Computes the ELiSH function.
+   *
+   * @param x Input data.
+   * @param y The resulting output activations.
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x, OutputVecType& y)
+  {
+    y.set_size(size(x));
+
+    for (size_t i = 0; i < x.n_elem; i++)
+      y(i) = Fn(x(i));
+  }
+
+  /**
+   * Computes the first derivatives of ELiSH function.
+   *
+   * @param y Input data.
+   * @return f'(x)
+   */
+  static double Deriv(const double y)
+  {
+    if (y < 0.0)
+    {
+      return std::exp(y) - 2 / (1 + std::exp(y)) + 
+                    2 / std::pow(1 + std::exp(y) , 2);  
+    }
+    return 1 / (1 + std::exp(-y)) + y * std::exp(-y) / 
+                         std::pow(1 + std::exp(-y) , 2);
```

Comment:
For y = f(x) , `Deriv` function takes in y as parameter and returns f'(x). Here you have calculated f'(x) taking x as input. I think you need to change the way you're calculating derivative.

Reply:
Actually You may be correct. First I hit the same issue. Then I look at how all activations are implemented and found Swish is written in the same way. Can you suggest a better way? Thanks.

Reply:
Sorry for the late reply. You can do it the same way as swish activation function for domain of x where it is the same. For the other domain of x you can try to see if f'(x) can be expressed in terms of f(x) that is y. Otherwise making a Inv() function to calculate Inv is what I would suggest

Reply:
I think we both are missing something. Why add Inverse If we can simply calculate the derivative of the function in terms of x. As u said earlier If I want to express f'(x) in terms of f(x) then only adding inverse makes sense. Else I think it will be faster to provide direct derivative in `Deriv` function.

Reply:
I think you're saying the opposite, if f'(x) can be expressed in terms of f(x) then we don't need inverse function however if that is not possible we need to calculate the inverse first.

Reply:
No no @prince776, `dy/dx = Inv(dx/dy).`
`y=f(x) so x= Inv(f)(y)`
Now `dy/dx = f'(x) & dx/dy = Inv(f)'(y)`
`So f'(x) = 1/( Inv(f)' (f(x)).`
I think this will clarify your issue. Thanks.